### PR TITLE
Gzip request body if server support it

### PR DESCRIPTION
**Pull request by PatrykGala**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

**Review by BartoszPrusak — DISMISSED** [review]

**Review by AleksanderWWW — DISMISSED** [review]

File: src/neptune/internal/backends/api_model.py [hunk 1]
```diff
@@ -132,6 +132,7 @@ class ClientConfig:
     _missing_features: FrozenSet[str]
```
Comment (Raalsky) [inline src/neptune/internal/backends/api_model.py:132]: Let's leave a note in the Changelog. That's performance improvement so it's at least beneficial from an user perspective.

File: src/neptune/internal/backends/hosted_client.py [hunk 2]
```diff
@@ -175,6 +189,10 @@ def create_backend_client(client_config: ClientConfig, http_client: HttpClient)
 
 @cache
 def create_leaderboard_client(client_config: ClientConfig, http_client: HttpClient) -> SwaggerClientWrapper:
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_client.py:191]: IDE warns me that `HttpClient` doesn't have `session` attribute. Probably we should change the typing to `RequestsClient`.

File: src/neptune/internal/backends/hosted_client.py [hunk 3]
```diff
@@ -66,6 +68,18 @@
 }
 
 
+class GzipAdapter(HTTPAdapter):
+    def send(self, request, stream=False, **kw):
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_client.py:72]: Can we add typings here?

File: src/neptune/internal/backends/hosted_client.py [hunk 4]
```diff
@@ -66,6 +68,18 @@
...
+        if request.body is not None and not stream:
+            request_body = request.body if isinstance(request.body, bytes) else bytes(request.body, "utf-8")
+            gzip_compress = zlib.compressobj(zlib.Z_DEFAULT_COMPRESSION, zlib.DEFLATED, zlib.MAX_WBITS | 16)
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_client.py:75]: Can we try catch that wrt. to `https://docs.python.org/3/library/zlib.html#zlib.error`?

File: src/neptune/internal/backends/hosted_client.py (hunk 3 above)
Comment (PatrykGala) [reply src/neptune/internal/backends/hosted_client.py]: done

File: src/neptune/internal/backends/hosted_client.py (hunk 2 above)
Comment (PatrykGala) [reply src/neptune/internal/backends/hosted_client.py]: done

File: src/neptune/internal/backends/api_model.py (hunk 1 above)
Comment (PatrykGala) [reply src/neptune/internal/backends/api_model.py:132]: done

File: src/neptune/internal/backends/hosted_client.py (hunk 4 above)
Comment (PatrykGala) [reply src/neptune/internal/backends/hosted_client.py:79]: I don't think it's necessary because an exception is very generic.

**Comment by PatrykGala** [comment]
https://github.com/neptune-ai/neptune-client/actions/runs/6403726256

File: src/neptune/internal/backends/hosted_client.py (hunk 4 above)
Comment (Raalsky) [reply src/neptune/internal/backends/hosted_client.py:79]: They mentioned that this could happen on compression as well

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,8 @@
 ## [UNRELEASED] neptune 1.7.1
 
+### Changes
```
Comment (Raalsky) [inline CHANGELOG.md:3]: You have to rebase as the latest version is `1.8.0`

File: src/neptune/internal/backends/hosted_client.py [hunk 6]
```diff
@@ -66,6 +72,18 @@
 }
 
 
+class GzipAdapter(HTTPAdapter):
+    def send(self, request: PreparedRequest, stream=False, **kw) -> Response:
```
Comment (Raalsky) [inline src/neptune/internal/backends/hosted_client.py:76]: `stream` needs a typing as well :wink:

Comment (PatrykGala) [reply src/neptune/internal/backends/hosted_client.py]: done

File: CHANGELOG.md (hunk 5 above)
Comment (PatrykGala) [reply CHANGELOG.md:6]: fixed

**Review by Raalsky — APPROVED** [review]

**Pull request merged**